app/code/Magento/Customer/Model/RedirectCookieManager.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Customer\Model;
+
+use Magento\Framework\Stdlib\Cookie\CookieMetadataFactory;
+use Magento\Framework\Stdlib\CookieManagerInterface;
+use Magento\Store\Api\Data\StoreInterface;
+
+/**
+ * Customer redirect cookie manager
+ * @SuppressWarnings(PHPMD.CookieAndSessionMisuse)
+ */
+class RedirectCookieManager
+{
+    const COOKIE_NAME = 'login_redirect';
+
+    /**
+     * @var CookieMetadataFactory
+     */
+    private $cookieMetadataFactory;
+
+    /**
+     * @var CookieManagerInterface
+     */
+    private $cookieManager;
+
+    /**
+     * @param CookieMetadataFactory $cookieMetadataFactory
+     * @param CookieManagerInterface $cookieManager
+     */
+    public function __construct(
+        CookieMetadataFactory $cookieMetadataFactory,
+        CookieManagerInterface $cookieManager
+    ) {
+        $this->cookieMetadataFactory = $cookieMetadataFactory;
+        $this->cookieManager = $cookieManager;
+    }
+
+    /**
+     * Get redirect route from cookie for case of successful login/registration
+     *
+     * @return null|string
+     */
+    public function getRedirectCookie()
+    {
+        return $this->cookieManager->getCookie(self::COOKIE_NAME, null);
+    }
+
+    /**
+     * Save redirect route to cookie for case of successful login/registration
+     *
+     * @param string $route
+     * @param StoreInterface $store
+     * @return void
+     */
+    public function setRedirectCookie($route, StoreInterface $store)
+    {
+        $cookieMetadata = $this->cookieMetadataFactory->createPublicCookieMetadata()
```

Comment:
Public cookie metadata is used. What will be the behavior and is it appropriate in the case if connection is secured?